File: SecondWave/joushin/1set/12789.cc
#include <iostream>
#include <stack>
int N;
int idx=1;
int input[1001];
using namespace std;

int main(){
	cin >> N;
	stack<int> S;
	int x;
	for(int i=1;i<=N;i++){
		cin >> input[i];
	}
	int i=1;
	while(1){
		if (input[i] == idx){
			i++;
			idx++;
		}
		else {
			if (!S.empty()&&  S.top() == idx){
				S.pop();
				idx++;
			}else{
				S.push(input[i]);
				i++;
			}
		}
		if (i == N+1)
			break;
	}
	while (!S.empty()){
		if (S.top() == idx)
		{
			S.pop();
			idx++;
		} 
		else {
			cout << "Sad\n";
			return (0);
		}
	}
	cout << "Nice\n";
}

File: SecondWave/sohlee/1set/12789.py
import sys
n=int(sys.stdin.readline())
li=list(map(int,sys.stdin.readline().split()))
stack=[]
i=1
j=0
while j<n:
    if li[j]==i:
        j+=1 ##바로 빼는 경우
        i+=1
        continue
    if len(stack)!=0:
        tmp=stack.pop()
        if tmp==i:
            i+=1
        else:
            stack.append(tmp)
            stack.append(li[j])
            j+=1
    else:
        stack.append(li[j])
        j+=1
if len(stack)!=0:
    while True:
        if i==n:
            break
        tmp=stack.pop()
        if tmp==i:
            i+=1
        else:
            break
if (i>=n):
    print("Nice")
else:
    print("Sad")